pgdb/v1/fts.go

```diff
@@ -108,6 +106,53 @@ func FullTextSearchVectors(docs []*SearchContent, additionalFilters ...jargon.Fi
 			_ = err
 		}
 	}
+	return rv
+}
+
+func camelSplitDocs(docs []*SearchContent) []lexeme {
+	rv := make([]lexeme, 0, 8)
+	camelCaseRegex := regexp.MustCompile(`[[:upper:]][[:lower:]]+`)
+	// acronymRegex - this is to catch acronyms in a named group 'token'
+	// the suffix after the group is foward looking to avoid the first letter of a camel cased word
+	// for example given the text AuthSNSTest this regex captures SNS in token but matches on SNSTe:
+	//
+	// Given the text `AuthSNSTest`:
+	// 		?P<token>[[:upper:]]+) -> SNS - captured in "token"
+	//		[[:upper:]][[:lower:]] -> Te - discarded
+	// Given the text `AuthSNS Foo`:
+	// 		?P<token>[[:upper:]]+) -> SNS - captured in "token"
+	//		\s -> the trailing empty space ' ' - discarded
+	// Given the text `AuthSNS`:
+	// 		?P<token>[[:upper:]]+) -> SNS - captured in "token"
+	//		$ -> end of string
+	acronymRegex := regexp.MustCompile(`(?P<token>[[:upper:]]+)([[:upper:]][[:lower:]]|$|\s)`)
+	template := "$token"
+	for _, doc := range docs {
+		if doc.Type == FieldOptions_FULL_TEXT_TYPE_ENGLISH_LONG {
+			continue
+		}
+		docValue := interfaceToValue(doc.Value)
+		for _, match := range camelCaseRegex.FindAllStringIndex(docValue, -1) {
+			result := docValue[match[0]:match[1]]
+			rv = append(rv, lexeme{strings.ToLower(result), match[0] + 1, doc.Weight})
+		}
+		for _, match := range acronymRegex.FindAllStringSubmatchIndex(docValue, -1) {
+			result := []byte{}
+			result = acronymRegex.ExpandString(result, template, docValue, match)
+			rv = append(rv, lexeme{strings.ToLower(string(result)), match[0] + 1, doc.Weight})
+		}
+	}
+	return rv
+}
+
+// FullTextSearchVectors converts a set of input documents
+// into a ::tsvector. Note: this function may generally ignore errors in input text, to be robust to
+// untrusted inputs, and will do its "best", for some value of "best".
+func FullTextSearchVectors(docs []*SearchContent, additionalFilters ...jargon.Filter) exp.Expression {
+	rv := make([]lexeme, 0, 8)
+
+	rv = append(rv, lemmatizeDocs(docs, additionalFilters...)...)
+	rv = append(rv, camelSplitDocs(docs)...)
```

Comment:
Do we need to uniqify these?

Reply:
yes

Reply:
well actually, i forgot this, but postgres's `:ts_vector` type does the uniqify automatically.

Reply:
I believe postgres does this automatically:

> Duplicate positions for the same lexeme are discarded.

https://www.postgresql.org/docs/current/datatype-textsearch.html